driver-core/src/main/com/mongodb/DBObjectCodec.java

```diff
@@ -338,14 +353,32 @@ private Object readValue(final BsonReader reader, final DecoderContext decoderCo
 
     private Object readBinary(final BsonReader reader, final DecoderContext decoderContext) {
         byte bsonBinarySubType = reader.peekBinarySubType();
+        Codec<?> codec;
 
         if (BsonBinarySubType.isUuid(bsonBinarySubType) && reader.peekBinarySize() == 16) {
```

Comment:
The `reader.peekBinarySize() == 16` is interesting. So if the BSON uuid is malformed with the wrong length Java will ignore it a return a Binary instead of raising an invalid bson error?

Reply:
Yes, that's correct.  The server doesn't report an error, so... should a driver?